framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/deployment/KieServerOperatorDeployment.java

```diff
@@ -64,15 +70,15 @@ public void waitForScale() {
         }
     }
 
-    private Server getAssociatedServerObject(KieApp kieApp) {
-        return Arrays.asList(kieApp.getSpec().getObjects().getServers()).stream().filter(s -> s.getName().equals(getServiceName()))
+    private Server getAssociatedServerObject(Spec spec) {
+        return Arrays.asList(spec.getObjects().getServers()).stream().filter(s -> s.getName().equals(getServiceName()))
                                                                                  .findAny()
-                                                                                 .orElseThrow(() -> new RuntimeException("Server with name " + getServiceName() + " not found. Available server names are: " + getAvailableServerNames(kieApp)));
+                                                                                 .orElseThrow(() -> new RuntimeException("Server with name " + getServiceName() + " not found. Available server names are: " + getAvailableServerNames(spec)));
     }
 
-    private String getAvailableServerNames(KieApp kieApp) {
-        String serverNames = Arrays.asList(kieApp.getSpec().getObjects().getServers()).stream().map(s -> s.getName())
+    private String getAvailableServerNames(Spec spec) {
```

Comment:
I think that here we can use kieApp.getStatus().getApplied() as we are getting available server names

Reply:
Spec comes from kieApp.getStatus().getApplied() already. See line 48 and 61.